src/Config/SystemConfigReader.ts

```diff
@@ -0,0 +1,153 @@
+import { logger } from '../Utils';
+import request, { RequestPromiseAPI } from 'request-promise';
+import { promises, constants } from 'fs';
+import { join } from 'path';
+import { homedir } from 'os';
+
+const { readFile, writeFile, access } = promises;
+const { F_OK } = constants;
+
+export interface SystemConfig {
+  sentryDsn?: string;
+}
+
+interface SystemConfigFile {
+  data: SystemConfig;
+  updatedAt: Date;
+}
+
+export class SystemConfigReader {
+  private readonly rotationInterval = 3600000;
+  private readonly path = join(homedir(), '.brightclirc');
+  private client: RequestPromiseAPI;
+
+  constructor(baseUrl: string) {
+    this.client = request.defaults({
+      baseUrl,
+      timeout: 1500,
+      json: true
+    });
+  }
+
+  public async read(): Promise<SystemConfig> {
+    await this.rotateIfNecessary();
+    const configFile = await this.getConfigFile();
+
+    return {
+      sentryDsn: process.env['SENTRY_DSN'],
+      ...configFile.data
+    };
+  }
+
+  private async rotateIfNecessary() {
```

Comment:
Split this method into two extracting a `shouldRotate`:

```ts
  private shouldRotate(): boolean {
    if (process.env.NODE_ENV !== 'production') {
      return false;
    }

    const configFile = this.getConfigFile();
    const lifeTime = Date.now() - configFile.updatedAt.getTime();
    return lifeTime >= this.rotationInterval;
  }
```

Reply:
Done. Not as elegant as suggested, since both methods depend on the same `configFile`